code/ai/aicode.cpp

```diff
@@ -14736,6 +14737,41 @@ static void ai_control_info_check(ai_info *aip, ship_info *sip, physics_info *pi
 		no_rot = true;
 	}
 
+	if (no_rot && no_lat) {
+		aip->ai_override_flags.reset();
+	}
+	else
+	{
+		if (aip->ai_override_flags[AI::Maneuver_Override_Flags::Dont_bank_when_turning] || sip->flags[Ship::Info_Flags::Dont_bank_when_turning])
+			AI_ci.control_flags |= CIF_DONT_BANK_WHEN_TURNING;
+		if (aip->ai_override_flags[AI::Maneuver_Override_Flags::Dont_clamp_max_velocity] || sip->flags[Ship::Info_Flags::Dont_clamp_max_velocity])
+			AI_ci.control_flags |= CIF_DONT_CLAMP_MAX_VELOCITY;
+		if (aip->ai_override_flags[AI::Maneuver_Override_Flags::Instantaneous_acceleration] || sip->flags[Ship::Info_Flags::Instantaneous_acceleration])
+			AI_ci.control_flags |= CIF_INSTANTANEOUS_ACCELERATION;
+	}
```

Comment:
If I'm reading this correctly, it looks like these ship_info flags will no longer be checked if there is no maneuver in progress.  This would break some existing functionality; for example ground vehicles shouldn't bank when turning even if they are following waypoints.

The AI control flags are intended to be set if there are maneuver flags *or* if the ship class entry has flags.

Reply:
This might have been buggy before then, as this code was only run for maneuvers even before this. I will double-check this behavior.

Reply:
I can confirm that `Instantaneous_acceleration` doesn't work on ships which aren't players or using maneuvers. This makes sense, since AI don't use CI, at least rotationally. For that, it would need to be in `ai_turn_towards_vector`, which `Dont_bank_when_turning` does correctly and still works. Laterally, however, there's no reason that it shouldn't work, and so should definitely be before the initial early return. (`Dont_clamp_max_velocity` also doesn't work, but is a bit of a weird one, AI never ask for more lateral speed than they have, but should this flag mean they should accelerate indefinitely as they are able???)

I would be in favor of these changes to make these flags work more generally for AI, but maybe not so late in the 'RC game' in this PR.

Reply:
Mmh.  Well, that's fair, especially as the mission for which I need these flags isn't released yet and probably won't be released before 22.4 is out either.  And so far I have only tested these flags on player-controlled ships.  I'll open a new issue.